pyfr/polys.py

```diff
@@ -10,7 +9,7 @@
 
 
 def jacobi(n, a, b, z):
-    j = [1]
+    j = [np.ones_like(z)]
 
     if n >= 1:
         j.append(((a + b + 2)*z + a - b) / 2)
```

Comment:
the remainder of this function is quite unclear. Is the single line saving of using `u,y = x,y` really worth the loss in clarity?
Also I think you calculate Jacobi polynomial without needing to use a three-term recession, so would remove some of the ifs. Here is a snippet from a Fortran implementation.
```
      J = 0.
      do m=0,n
         c = choose(n, m)*gamma(a + b + n + m + 1)/gamma(a + m + 1)
         J = J + c*((0.5*(x-1))**m)
      enddo
      J = J*gamma(a+n+1)/gamma(a+b+n+1)/fac(n)
```

Reply:
The function returns a list of all of the Jacobi polynomials from 0..n.  This works nicely as whenever we need the n'th we also need all of the lower degree polynomials too.  Thanks to three term recurrence relation these are effectively free.  It is also reasonably numerically stable and, as a minor benefit, avoids the need to evaluate any gamma functions.

However, due to the recurrence the conditionals are required to correctly handle start-up.

Reply:
Fair enough.